### PR TITLE
Configurable zoom mode setting

### LINKED ISSUE #192: Extension changes Background Adjustment
On photo change, Background Adjustment is reset to Zoom.
This extension shouldn't alter any existing setting, for example Scaled.

It seems to be from:
            backgroundSettings.setString('picture-options', 'zoom');
            
This should be a setting left to the individual not hardcoded in the extension.

Other than this, great app

### PR DESCRIPTION
**Pull request by Lucki**
Fix #192

Maybe there's better wording for zoom mode. Gnomes own 'picture-options' is somewhat meaningless though.

### CODE REVIEW

File: src/prefs.ts [hunk 1]
```diff
@@ -71,6 +71,13 @@ class RandomWallpaperSettings extends ExtensionPreferences {
             'selected',
             Gio.SettingsBindFlags.DEFAULT);
 
+        const comboZoomMode = this._getAs<Adw.ComboRow>(builder, 'combo_zoom_mode');
+        comboZoomMode.model = Gtk.StringList.new(WallpaperManager.getZoomModeNameList());
+        settings.bind('zoom-mode',
```
Comment (ifl0w) [inline src/prefs.ts:76]: I'd then also directly bind to the picture-options for background/screensaver.

File: src/manager/defaultWallpaperManager.ts [hunk 2]
```diff
@@ -54,8 +54,7 @@ class DefaultWallpaperManager extends WallpaperManager {
             // merged wallpapers need mode "spanned"
             backgroundSettings.setString('picture-options', 'spanned');
         else
-            // single wallpapers need mode "zoom"
-            backgroundSettings.setString('picture-options', 'zoom');
+            backgroundSettings.setString('picture-options', getZoomName(new Settings().getInt('zoom-mode')));
```
Comment (ifl0w) [inline src/manager/defaultWallpaperManager.ts:57]: The downside is that we couldn't reset to the previously selected state here without duplicating the option in our schema.

**Review by ifl0w — COMMENTED** [review]
Thanks for the PR/MR! I placed a few comments, but I'll have to think about this change a bit more. Feel free to add your opinions.

The two things on my mind are:
* Should we store/replicate the picture-options value in our schema? I tend to no, but see my other comments for further thoughts.
* Should we expose an option to change the background color as well (relevant for centered/scaled)? I guess, we can address that in a separate PR anyway, so this is not so important here.

**Comment by Lucki** [comment]
> * Should we store/replicate the picture-options value in our schema? I tend to no, but see my other comments for further thoughts.

As a user I wouldn't expect my setting to change randomly by using other programs. So I think we should save and apply our own value.
And similarly we have to tie the multi monitor setting somehow to the scaling setting to get back out of the `spanned` mode.

Also saving this setting separately could be beneficial if we implement HydraPaper's `--modes [mode, mode…]` option. See #195

File: src/manager/wallpaperManager.ts [hunk 3]
```diff
@@ -104,8 +125,65 @@ function getModeNameList(): string[] {
...
+ * Get a list of human readable enum entries.
+ *
+ * @returns {string[]} Array with key names
+ */
+function getZoomModeNameList(): string[] {
+    const list: string[] = [];
+
+    const values = Object.values(ZoomMode).filter(v => !isNaN(Number(v)));
+    for (const i of values)
+        list.push(getZoomName(i as ZoomMode));
```
[the remark this answers is not in the document]
Comment (Lucki) [reply src/manager/wallpaperManager.ts:178]: Sure, I already had that in the early TS days for our own enums when they were still in the gsettings. I'll add it back in.

How do think about also dropping `wallpaper`, `stretched` and `spanned` as user choices, as they ignore the image aspect ratio and I've never came across a tiled wallpaper?

[the remark this answers is not in the document]
Comment (ifl0w) [reply src/manager/wallpaperManager.ts]: Thanks! Changes look much leaner now.

[the remark this answers is not in the document]
Comment (ifl0w) [reply src/manager/wallpaperManager.ts]: > How do think about also ignoring wallpaper, stretched and spanned as user choices, as they ignore the image aspect ratio and I've never came across a tiled wallpaper?

Having similar feelings about those. So we can filter them out for now.

**Comment by ifl0w** [comment]
> As a user I wouldn't expect my setting to change randomly by using other programs. So I think we should save and apply our own value.

You have a point there. We probably also should reset the zoom mode to the default value when disabling the extension.

> Not sure about this one. The random wallpapers will make the color always be incorrect. Best would be a neutral color to adapt to as many wallpapers as possible. We could also hardcode such a color, e.g. black 🤷 

I think, that it should be a user's choice what color they want to use, but let's address this separately.

**Review by ifl0w — APPROVED** [review]
Looks good to me now! Thanks for this feature!

I have a commit locally for resetting the picture-options value and will probably add that to #198 and request feedback there.

**Pull request merged**